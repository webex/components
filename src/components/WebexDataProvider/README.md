# Webex Activity Data Provider Component

LALLI HELP ME HERE PLEASE. MY BRAIN IS FRIED RIGHT NOW!
