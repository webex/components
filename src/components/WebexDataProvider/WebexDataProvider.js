import React from 'react';
import PropTypes from 'prop-types';

export const AdapterContext = React.createContext();

export default function WebexDataProvider({adapter, children}) {
  return <AdapterContext.Provider value={adapter}>{children}</AdapterContext.Provider>;
}

WebexDataProvider.propTypes = {
  adapter: PropTypes.shape({
    activitiesAdapter: PropTypes.object.isRequired,
    meetingsAdapter: PropTypes.object.isRequired,
    peopleAdapter: PropTypes.object.isRequired,
    roomsAdapter: PropTypes.object.isRequired,
  }).isRequired,
  children: PropTypes.node.isRequired,
};
