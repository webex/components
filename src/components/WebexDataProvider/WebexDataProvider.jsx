/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';

import {AdapterContext} from '../hooks';

/**
 * Provides an adapter context to the wrapped component.
 *
 * @param {object} props  Data passed to the provider
 * @param {object} props.adapter  Adapter for context
 * @param {React.Component} props.children  Component children to wrap
 * @returns {React.Component} Component with access to the adapter context
 */
export default function WebexDataProvider({adapter, children}) {
  return <AdapterContext.Provider value={adapter}>{children}</AdapterContext.Provider>;
}

WebexDataProvider.propTypes = {
  adapter: PropTypes.shape({
    activitiesAdapter: PropTypes.object.isRequired,
    meetingsAdapter: PropTypes.object.isRequired,
    membershipsAdapter: PropTypes.object.isRequired,
    peopleAdapter: PropTypes.object.isRequired,
    roomsAdapter: PropTypes.object.isRequired,
  }).isRequired,
  children: PropTypes.node.isRequired,
};
