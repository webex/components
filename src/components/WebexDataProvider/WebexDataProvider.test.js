import React from 'react';

import WebexDataProvider from './WebexDataProvider';

describe('Webex Data Provider component', () => {
  const adapter = {
    roomsAdapter: {},
    activitiesAdapter: {},
    peopleAdapter: {},
  };

  test('matches snapshot', () => {
    const component = shallow(
      <WebexDataProvider adapter={adapter}>
        <div className="test" />
      </WebexDataProvider>
    );

    expect(component).toMatchSnapshot();
  });
});
