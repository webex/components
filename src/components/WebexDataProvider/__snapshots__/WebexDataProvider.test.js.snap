// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webex Data Provider component matches snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "activitiesAdapter": Object {},
      "peopleAdapter": Object {},
      "roomsAdapter": Object {},
    }
  }
>
  <div
    className="test"
  />
</ContextProvider>
`;
